File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetVideoListUseCase.java
package com.hwonchul.movie.domain.usecase.details;

import com.hwonchul.movie.domain.model.Video;

import java.util.List;

import io.reactivex.rxjava3.core.Flowable;

/**
 * 영화 예고편 가져오기 UseCase
 */
public interface GetVideoListUseCase {

    Flowable<List<Video>> invoke(int movieId);
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetVideoListUseCase.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.model.Video
import kotlinx.coroutines.flow.Flow

/**
 * 영화 예고편 가져오기 UseCase
 */
interface GetVideoListUseCase {

    suspend operator fun invoke(movieId: Int): Flow<Result<List<Video>>>
}